pandas/_libs/lib.pyx

```diff
@@ -1913,8 +1913,8 @@ def maybe_convert_objects(ndarray[object] objects, bint try_float=0,
         ndarray[int64_t] ints
         ndarray[uint64_t] uints
         ndarray[uint8_t] bools
-        ndarray[int64_t] idatetimes
-        ndarray[int64_t] itimedeltas
+        int64_t[::1]  idatetimes
+        int64_t[::1] itimedeltas
```

Comment:
compiler compalins about dim_info[0].strided being possibly uninited. I don't fully understand why, but using memory views generates different code and should be interchangeable.

Reply:
Makes sense conceptually. @jbrockmendel has good experience in converting between these though so may have insight to offer

Reply:
IIRC this function won't return idatetimes or itimedeltas to the user, so this should be fine.  Is there a particular reason for using `::1` instead of just `:`?  We generally do the latter.

Reply:
@pilkibun if you'd change this to ``[:]`` otherwise this PR lgtm.

Reply:
That effectively means we expect `np.empty` to return a c-contig array. I'll lose it.